File: backend/app/routers/colleges.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.database import get_db
from app.schemas.college import CollegeOut
from app.services import catalog_service

router = APIRouter(tags=["colleges"])


@router.get("/colleges", response_model=list[CollegeOut])
def list_colleges(db: Session = Depends(get_db)) -> list[CollegeOut]:
    """Return every college/school, so a client can ask "which school?" before
    narrowing the 147-program catalog down to one program."""
    return catalog_service.list_colleges(db)

File: frontend/src/lib/api/colleges.ts
import { apiFetch } from "@/lib/api/client"
import type { CollegeOut } from "@/lib/types"

/** Fetch every college/school, for the wizard's "which school?" first step. */
export function listColleges(): Promise<CollegeOut[]> {
  return apiFetch<CollegeOut[]>("/colleges")
}
